File: domain/src/main/java/com/ahn/domain/exchange/usecase/RefreshExchangeRatesUseCase.kt
package com.ahn.domain.exchange.usecase

import com.ahn.domain.exchange.repository.ExchangeRateRepository
import javax.inject.Inject

class RefreshExchangeRatesUseCase @Inject constructor(private val repository: ExchangeRateRepository) {
    suspend operator fun invoke() {
        repository.refreshExchangeRates()
    }
}

File: ios/Packages/Domain/Sources/Domain/Exchange/UseCase/RefreshExchangeRatesUseCase.swift
public struct RefreshExchangeRatesUseCase: Sendable {
    private let repository: any ExchangeRateRepository

    public init(repository: any ExchangeRateRepository) {
        self.repository = repository
    }

    public func callAsFunction() async throws {
        try await repository.refreshExchangeRates()
    }
}
